test/spec/Feature/Auth/AuthSpec.hs

```diff
@@ -159,3 +159,14 @@ spec = describe "authorization" $ do
       `shouldRespondWith` 200
     request methodGet "/authors_only" [authHeader "bearer" token] ""
       `shouldRespondWith` 200
+
+  describe "when JWT is empty string" $ do
+
+    it "raises error on empty token in Auth header" $ do
+      let auth = authHeaderJWT ""
+      request methodGet "/authors_only"
+        [auth]
+        ""
+        `shouldRespondWith`
+        [json| {"code":"PGRST301","details":null,"hint":null,"message":"JWSError (CompactDecodeError Invalid number of parts: Expected 3 parts; got 2)"} |]
+        { matchStatus = 401 }
```

Comment:
This is a change in behavior from previous versions?

On `main` right now I'm doing:

```
$ curl localhost:3000/authors_only -H "Authorization: Bearer"
{"code":"42501","details":null,"hint":null,"message":"permission denied for table authors_only"}

$ curl localhost:3000/authors_only -H "Authorization: Bearer "
{"code":"42501","details":null,"hint":null,"message":"permission denied for table authors_only"}
```

Is that equivalent to the above test?

Reply:
> This is a change in behavior from previous versions?

If so, I think this is an improvement. Users could have been passing an empty JWT by accident and getting the db permission error instead of a decoding error.

> JWSError (CompactDecodeError Invalid number of parts: Expected 3 parts; got 2)

The above is a mysterious error, I wonder if it could be improved.

Reply:
> If so, I think this is an improvement.

Yes, it is an improvement.

> The above is a mysterious error, I wonder if it could be improved.

We have an improvement issue tracked here (#3600). But for now we can improve this one. How about?

```haskell
"JWSError (CompactDecodeError Invalid number of parts: Expected 3 parts in the JWT)"
```

Reply:
> Yes, it is an improvement.

Ok cool. So then this is not really tied to the JWT cache issue and it is an important fix. How about a new focused PR for the "empty JWT case giving a good error message"?

Reply:
> How about a new focused PR for the "empty JWT case giving a good error message"?

Sure, as soon as this gets merged, I'll open the other PR.

Reply:
Also, this bad error message isn't just for "empty" JWT. We get the same for `Authorization: Bearer abc`. So a PR just for empty JWT would be enough.

Reply:
@steve-chavez  Since we are doing #3600 anyway, lets close this PR and move it to the aforementioned PR?

Reply:
> Since we are doing https://github.com/PostgREST/postgrest/issues/3600 anyway, lets close this PR and move it to the aforementioned PR?

Yes, sounds good.